### PR TITLE
Replaced random.cat with the TheCatAPI

### PR DESCRIPTION
**Pull request by CaldeiraG**
Since https://aws.random.cat/meow was taken down, I replaced the code using TheCatAPI.

I'm not too sure if I need any more error handling within the GET request or not since it only checks for `length`

### CODE REVIEW

File: comment_scan/replier.rb [hunk 1]
```diff
@@ -336,14 +336,13 @@ def bad_command(msg_id, parent_id, chat_user, room_id, *args)
...
-        case cat_response.code
-            when 200 #All good!
-                @chatter.say(":#{msg_id} #{cat_response.parsed_response["file"]}", room_id)
-            when 404
-                @chatter.say("O noes! Cats not found!", room_id)
-            when 500...600
-                @chatter.say("ZOMG ERROR #{cat_response.code}...and no cats :(", room_id)
+        cat_response = HTTParty.get("https://api.thecatapi.com/v1/images/search", :headers => { 'content-type': 'application/json', 'x-api-key': '0605155b-c002-48bf-a3ab-45be10448c44' }) #api-key is necessary to fetch image
+        #@chatter.say(cat_response.parsed_response[0]);
+        if cat_response.length > 0
```
Comment (scohe001) [inline comment_scan/replier.rb:341]: This is a good sanity check to make sure that the API is actually giving us something when it says it is, but even better would be to check the actual response code--this'll tell you if the site is down (404), you encounter some kind of server error (500's) or if the API claims it actually gave you good data (200). You can do that the same way I was before (by checking `cat_response.code`).

File: comment_scan/replier.rb [hunk 2]
```diff
@@ -336,14 +336,13 @@ def bad_command(msg_id, parent_id, chat_user, room_id, *args)
...
 
-        cat_response = HTTParty.post("https://aws.random.cat/meow")
-        case cat_response.code
-            when 200 #All good!
-                @chatter.say(":#{msg_id} #{cat_response.parsed_response["file"]}", room_id)
-            when 404
-                @chatter.say("O noes! Cats not found!", room_id)
-            when 500...600
-                @chatter.say("ZOMG ERROR #{cat_response.code}...and no cats :(", room_id)
+        cat_response = HTTParty.get("https://api.thecatapi.com/v1/images/search", :headers => { 'content-type': 'application/json', 'x-api-key': '0605155b-c002-48bf-a3ab-45be10448c44' }) #api-key is necessary to fetch image
```
Comment (scohe001) [inline comment_scan/replier.rb]: Is the `content-type` tag required in the header here? I would assume HTTParty would default to a json type. Have you tested without that?

**Comment by CaldeiraG** [comment]
Hey, sorry for the delay.

It seems like random.cat is now back on so I don't know if it would make sense to move forward with this PR. I will still use TheCatAPI on my [local instance](https://chat.stackexchange.com/users/472590/caldeirabot) but let me know.

**Comment by scohe001** [comment]
Sounds good. In that case, I'll close this PR.

**Pull request closed without merging**